libzfs.pyx

```diff
@@ -3261,6 +3261,31 @@ cdef class ZFSResource(ZFSObject):
         finally:
             free(iter.array)
 
+    def update_properties(self, all_properties):
+        cdef NVList props = NVList()
+        cdef int ret
+        invalid_values = []
+        for prop_name, prop_details in all_properties.items():
+            cur_prop = self.properties.get(prop_name)
+            if prop_details.get('source') == 'INHERIT' and cur_prop:
+                cur_prop.inherit(recursive=prop_details.get('recursive', False))
+            else:
+                if 'value' in prop_details:
+                    props[prop_name] = prop_details['value']
+                elif 'parsed' in prop_details:
+                    props[prop_name] = serialize_zfs_prop(prop_name, prop_details['parsed'])
```

Comment:
Can parsing fail?

Reply:
We are handling errors while parsing and this should ideally not. To clarify, this behavior is consistent with what is happening right now when we update properties and we have not seen issues of parsing failing.